kong/runloop/handler.lua

```diff
@@ -961,51 +967,87 @@ return {
         end
       end
 
-      if strategy ~= "off" then
+      do  -- start some rebuild timers
         local worker_state_update_frequency = kong.configuration.worker_state_update_frequency or 1
 
-        local function rebuild_timer(premature)
+        local router_async_opts = {
+          name = "router",
+          timeout = 0,
+          on_timeout = "return_true",
+        }
+
+        local function rebuild_router_timer(premature)
           if premature then
             return
           end
 
-          local router_update_status, err = rebuild_router({
-            name = "router",
-            timeout = 0,
-            on_timeout = "return_true",
-          })
-          if not router_update_status then
+          -- Don't wait for the semaphore (timeout = 0) when updating via the
+          -- timer.
+          -- If the semaphore is locked, that means that the rebuild is
+          -- already ongoing.
+          local ok, err = rebuild_router(router_async_opts)
+          if not ok then
             log(ERR, "could not rebuild router via timer: ", err)
           end
+        end
 
-          local plugins_iterator_update_status, err = rebuild_plugins_iterator({
-            name = "plugins_iterator",
-            timeout = 0,
-            on_timeout = "return_true",
-          })
-          if not plugins_iterator_update_status then
+        local _, err = kong.timer:named_every("router-rebuild",
```

Comment:
Is this change related to incremental sync? If so, how? This does spin a recurring timer to rebuild the router? Shouldn't it rebuilt based on reconfiguration events, like it did before?